numba/targets/numbers.py

```diff
@@ -1352,3 +1353,41 @@ def constant_complex(context, builder, ty, pyval):
 def constant_integer(context, builder, ty, pyval):
     lty = context.get_value_type(ty)
     return lty(pyval)
+
+
+#-------------------------------------------------------------------------------
+# Reinterpretation of scalars with alternate types, the `view` method.
+
+@intrinsic
+def viewer(tyctx, val, viewty):
+    bits = val.bitwidth
+    if isinstance(viewty.dtype, types.Integer):
+        bitcastty = ir.IntType(bits)
+    elif isinstance(viewty.dtype, types.Float):
+        bitcastty = ir.FloatType() if bits == 32 else ir.DoubleType()
+    else:
+        raise (".view() argument must be an integer or float type")
```

Comment:
there's an exception class missing here

Reply:
good catch!

Reply:
Now we wait for CI to complete.

Reply:
that it wasn't caught by CI suggests it wasn't tested, perhaps add the test?

Reply:
Absolutely.